### DISCUSSION TITLE
How to create custom surfaces/materials

### DISCUSSION THREAD

**Discussion by Nolram12345** [google/filament#7718] [Q&A] [answered]
Hello! I'm curious on how one would go about adding custom surface shaders/materials in filament. I couldn't find any documentation on it (though to be honest, I have not looked through all the source yet) - is there any guidance on it?
The readme does state _Custom lighting/surface shading_ as a feature.

**Comment by romainguy** [comment, marked as answer]
See the docs: https://google.github.io/filament/Materials.html#customsurfaceshading

**Reply by Nolram12345** [reply]
Thanks, I missed that!
